src/lib/ensemble/bagging.ts

```diff
@@ -0,0 +1,207 @@
+/**
+ * References:
+ * - https://scikit-learn.org/stable/modules/generated/sklearn.ensemble.BaggingClassifier.html
+ */
+import { DecisionTreeClassifier } from '../tree';
+import { Type1DMatrix, Type2DMatrix } from '../types';
+import math from '../utils/MathExtra';
+import { ensure2DMatrix, inferShape } from '../utils/tensors';
+import { validateFitInputs } from '../utils/validation';
+
+/**
+ * A Bagging classifier is an ensemble meta-estimator that fits
+ * base classifiers each on random subsets of the original dataset
+ * and then aggregate their individual predictions by voting
+ * to form a final prediction
+ * @example
+ * const classifier = new BaggingClassifier({
+ *  baseEstimator: LogisticRegression,
+ *  maxSamples: 1.0,
+ * });
+ * const X = [[1], [2], [3], [4], [5]];
+ * const y = [1, 1, 1, 1, 1];
+ * classifier.fit(X, y);
+ * classifier.predict(X);
+ */
+export class BaggingClassifier {
+  private BaseEstimator: any;
```

Comment:
I think we can correctly type this later on. We have `IMlModel` interface but it's not really useful right now. I can work on this.

Reply:
Opened an issue for this https://github.com/machinelearnjs/machinelearnjs/issues/245